File: Java/Solutions/3.2. Simple Conditions - Exam Problems/src/SleepyCatTom.java
import java.util.Scanner;

public class SleepyCatTom {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int holidays = Integer.parseInt(scanner.nextLine());

        int workingDays = 365 - holidays;

        int totalPlayMinutes = holidays * 127 + workingDays * 63;

        double difference = Math.abs(totalPlayMinutes - 30000);
        double hours = difference / 60;
        double minutes = difference % 60;

        if (totalPlayMinutes > 30000) {
            System.out.println("Tom will run away");
            System.out.printf("%.0f hours and %.0f minutes more for play",
                    Math.floor(hours), Math.floor(minutes));
        } else {
            System.out.println("Tom sleeps well");
            System.out.printf("%.0f hours and %.0f minutes less for play",
                    Math.floor(hours), Math.floor(minutes));
        }
    }
}

File: JavaScript/Solutions/03.Simple-conditions-exam-problems/SleepyCatTom.js
function sleepyCatTom([holidays]) {

    let workingDays = 365 - holidays;

    let totalPlayMinutes = holidays * 127 + workingDays * 63;

    let difference = Math.abs(totalPlayMinutes - 30000);
    let hours = difference / 60;
    let minutes = difference % 60;

    if (totalPlayMinutes > 30000) {
        console.log("Tom will run away");
        console.log(`${Math.floor(hours)} hours and ${Math.floor(minutes)} minutes more for play`);
    }
    else {
        console.log("Tom sleeps well");
        console.log(`${Math.floor(hours)} hours and ${Math.floor(minutes)} minutes less for play`);
    }
}
